### PR TITLE
Wip/cragw/mediatek scaler fix and vcp c8

### PR DESCRIPTION
**Pull request by CragW**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/mediatek-scaler/fu-mediatek-scaler-device.c [hunk 1]
```diff
@@ -402,31 +436,46 @@ fu_mediatek_scaler_device_probe(FuDevice *device, GError **error)
...
-	if (!fu_device_probe(FU_DEVICE(udev_parent), error))
+	/* prioritize DDC/CI commands in display controller */
+	if (!fu_mediatek_scaler_device_set_ddc_priority(device, FU_DDCCI_PRIORITY_UP, error))
 		return FALSE;
 
-	vid = g_strdup_printf("%04X", fu_udev_device_get_subsystem_vendor(udev_parent));
-	pid = g_strdup_printf("%04X", fu_udev_device_get_subsystem_model(udev_parent));
+	/* mediatek display is connected */
+	if (!fu_mediatek_scaler_display_is_connected(device, error)) {
+		fu_mediatek_scaler_device_set_ddc_priority(device, FU_DDCCI_PRIORITY_NORMAL, error);
```
Comment (CragW) [inline plugins/mediatek-scaler/fu-mediatek-scaler-device.c:464]: in contrast to add this in every false returns, any better way to run a code when probe leaves?

**Comment by CragW** [comment]
@GregLo007

File: plugins/mediatek-scaler/fu-mediatek-scaler-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c:464]: I think we want to clean it up something like what's in wip/hughsie/mediatek-weird -- you're relying on a lot of daemon auto-fixup at the moment, e.g. refcounting the opens of the i2c device for instance.

Comment (hughsie) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c:464]: Let me rebase this onto  wip/hughsie/mediatek-weird -- then I think it'll be a lot nicer.

Comment (hughsie) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c:464]: @CragW I've merged that functionality on top of wip/hughsie/mediatek-weird -- can you test that branch for me please.

File: plugins/mediatek-scaler/fu-mediatek-scaler-device.c [hunk 2]
```diff
@@ -402,31 +436,46 @@ fu_mediatek_scaler_device_probe(FuDevice *device, GError **error)
...
 		return FALSE;
 
-	/* set hardware version */
-	hw_ver = fu_mediatek_scaler_device_get_hardware_version(device, error);
-	if (hw_ver == NULL)
+	vid_u16 = fu_udev_device_get_subsystem_vendor(udev_parent);
+	pid_u16 = fu_udev_device_get_subsystem_model(udev_parent);
+
+	/* supported host system only */
+	if (!(vid_u16 != 0x1028) && ((pid_u16 != 0x0c88) || (pid_u16 == 0x0c8a)))
```
Comment (superm1) [inline plugins/mediatek-scaler/fu-mediatek-scaler-device.c:451]: You need to set error here.

Comment (CragW) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c:451]: > You need to set error here.

Thanks for reviewing here, the vcp test has been mreged to another brach. I will close this PR.

**Pull request closed without merging**